packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js

```diff
@@ -78,7 +86,7 @@ function InteractionLayer ({
     }
   }
 
-  function onFinish (event, node) {
+  function onFinish (event = {}, node) {
```

Comment:
Yes, this will fail without a default parameter because of the line underneath it, `const { target, pointerId } = event`. This function will be called without an event on the `useEffect` change made in this PR.

Reply:
`onFinish` is an event handler that's passed as a prop to individual mark components and then called when you finish drawing. https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/plugins/drawingTools#react-components

Here's a couple of examples where it's passed a pointer event:

https://github.com/zooniverse/front-end-monorepo/blob/661bc16f39297d1eb5fc4bf5555515d973c0ac68/packages/lib-classifier/src/plugins/drawingTools/components/Point/Point.js#L25-L27

https://github.com/zooniverse/front-end-monorepo/blob/661bc16f39297d1eb5fc4bf5555515d973c0ac68/packages/lib-classifier/src/plugins/drawingTools/components/Ellipse/Ellipse.js#L25-L27

`useEffect` here is using `onFinish` in a slightly different manner than it was originally intended to be used, but I'm not sure if that's really a big deal, or if the original idea of it as an event handler is useful when we're dealing with multiple gestures to create a shape.

Reply:
I would suggest that we have an `onPointerUp` event handler and a separate `onFinish` function and decouple what's going on here. That's out of scope for this fix though.

Reply:
I think I might have chosen bad examples that both use `onPointerUp` to finish a mark. `onFinish` was added as a Mark prop so that we have the decoupling that @srallen mentions. Finishing an Ellipse on pointerUp was definitely a mistake, because you really want to wait until both axes have been drawn before finishing the shape.

Prior to that, the interaction layer had an `onPointerUp` handler, but then the drawing surface is dictating how every mark, regardless of type, should handle events. That breaks when you want a mark that doesn't conform to the generic mark behaviour eg. a line that's drawn by tapping twice on the drawing canvas.

Reply:
I think the piece that still needs to be decoupled is setting and releasing pointer capture for multiple gestures. I don’t think that needs to happen in this PR, but `onFinish()` in this code will not manually release a pointer, which means this PR relies on the browser to release any captured pointers. 

I think `setPointerCapture` gets called during the initial pointerDown. Browsers should release it (I think) on the next pointerUp emitted with that pointerID.

Reply:
I think generally speaking that this is all a bit too abstracted. What I'd like to see is that each tool has its own set of pointer down, pointer up, etc handlers and yes, I realize that this might mean some duplication of code, but we'll buy back some flexibility and clarity about what each tool does for each event. Then we can decide on how to abstract it to some general handlers if needed. I'd also like to see it consolidated a bit: either let's define these in the models or let's define them in the components. Right now we have pointer event handling spread across MST models and components and again, it's not clear what gets called when. Out of scope for this though.

Reply:
Sure. I was more concerned that this block isn't being called so wanted to be absolutely sure we aren't capturing the pointer then not releasing it.
https://github.com/zooniverse/front-end-monorepo/blob/f0f63edeafaabb7c7126197fc3def7194b4c2ada/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js#L96-L98